### DISCUSSION TITLE
Is there an option to allow only mediacms members view uploaded media

### DISCUSSION THREAD

**Discussion by behnam-marandi** [mediacms-io/mediacms#422] [Q&A] [answered]
PORTAL_WORKFLOW allows to set media access to public(anyone), private(only owner, admins and editor) or unlisted.

Is there an option to set media access to all member users of the MediaCMS? Like anyone who is logged in could see media and categories with all included media in each category.

**Comment by mgogoulos** [comment, marked as answer]
not sure whether this helps you, perhaps

- make the portal workflow public, but at the same time set `GLOBAL_LOGIN_REQUIRED = True` so that only logged in users can see content
- checkout options on django-allauth settings that affect registration: https://github.com/mediacms-io/mediacms/blob/main/cms/settings.py
Eg set the portal invite only, or set email confirmation as mandatory, so that you control who registers

If you end up with something that works for your setup, consider writing a few words on the documentation too!

https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md

**Comment by behnam-marandi** [comment]
Thanks @mgogoulos 
It works for me.
Created the pull request to apply instructions to admins_docs: https://github.com/mediacms-io/mediacms/pull/424

**Comment by behnam-marandi** [comment]
I am not sure if these changes affect password reset or there are some other configurations that affect this.

When I try the 'Forget Password' page redirects to /accounts/password/reset but load 'Sign In' page.

`ACCOUNT_LOGIN_ON_PASSWORD_RESET` is set to `True`.

**Comment by shodanx2** [comment]
I have reviewed 

https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md
https://github.com/mediacms-io/mediacms/blob/main/cms/settings.py

And I don't see how the invite only mode can be enabled ?

Or, alternatively, how can the registrations can be made pending administrator's review after the email is verified ?

For controlling the registrations I see

you can ban certain email domains
you can disable registration 
you can make the portal invite only (how?)
and "set email confirmation as mandatory, so that you control who registers" (how does that stop anyone with a valid address from registering ?)

thanks !
